recipes/armadillo/all/patches/0002-Guard-dependency-discovery-10.7.x.patch

```diff
@@ -0,0 +1,152 @@
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 7857f8c..00d80b0 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -274,7 +274,13 @@ if(APPLE)
+   set(ARMA_USE_ACCELERATE true)
+   
+   if(ALLOW_OPENBLAS_MACOS)
+-    include(ARMA_FindOpenBLAS)
++    if(USE_SYSTEM_OPENBLAS)
++        include(ARMA_FindOpenBLAS)
++    elseif(USE_OPENBLAS)
++        find_package(OpenBLAS)
++    else()
++        set(OpenBLAS_FOUND NO)
++    endif()
+     message(STATUS "OpenBLAS_FOUND = ${OpenBLAS_FOUND}")
+     message(STATUS "")
+     message(STATUS "*** If use of OpenBLAS is causing problems,")
+@@ -289,8 +295,16 @@ if(APPLE)
+   endif()
+   
+   if(ALLOW_BLAS_LAPACK_MACOS)
+-    include(ARMA_FindBLAS)
+-    include(ARMA_FindLAPACK)
++    if(USE_SYSTEM_BLAS)
++      include(ARMA_FindBLAS)
++    else()
++      set(BLAS_FOUND NO)
++    endif()
++    if(USE_SYSTEM_LAPACK)
++      include(ARMA_FindLAPACK)
++    else()
++      set(LAPACK_FOUND NO)
++    endif()
+     message(STATUS "  BLAS_FOUND = ${BLAS_FOUND}"  )
+     message(STATUS "LAPACK_FOUND = ${LAPACK_FOUND}")
+     message(STATUS "")
+@@ -329,14 +343,44 @@ if(APPLE)
+   
+ else()
+   
+-  include(ARMA_FindMKL)
+-  include(ARMA_FindOpenBLAS)
+-  include(ARMA_FindATLAS)
+-  include(ARMA_FindBLAS)
+-  include(ARMA_FindLAPACK)
++  if(USE_SYSTEM_MKL)
++    include(ARMA_FindMKL)
++  else()
++    set(MKL_FOUND NO)
++  endif()
++
++  if(USE_SYSTEM_OPENBLAS)
++    include(ARMA_FindOpenBLAS)
++  elseif(USE_OPENBLAS)
++    find_package(OpenBLAS)
++  else()
++    set(OpenBLAS_FOUND NO)
++  endif()
++
++  if(USE_SYSTEM_ATLAS)
++    include(ARMA_FindATLAS)
++  else()
++    set(ATLAS_FOUND NO)
++  endif()
++
++  if(USE_SYSTEM_BLAS)
++    include(ARMA_FindBLAS)
++  else()
++    set(BLAS_FOUND NO)
++  endif()
++
++  if(USE_SYSTEM_LAPACK)
++    include(ARMA_FindLAPACK)
++  else()
++    set(LAPACK_FOUND NO)
++  endif()
+   
+   if(ALLOW_FLEXIBLAS_LINUX AND (${CMAKE_SYSTEM_NAME} MATCHES "Linux"))
+-    include(ARMA_FindFlexiBLAS)
++    if(USE_SYSTEM_FLEXIBLAS)
++      include(ARMA_FindFlexiBLAS)
++    else()
++      set(FlexiBLAS_FOUND NO)
++    endif()
+   endif()
+   
+   message(STATUS "      MKL_FOUND = ${MKL_FOUND}"       )
+@@ -470,7 +514,12 @@ if(DETECT_HDF5)
+     # HDF5_INCLUDE_DIRS is the correct include directory.  So, in either case we
+     # can use the first element in the list.  Issue a status message, too, just
+     # for good measure.
+-    list(GET HDF5_INCLUDE_DIRS 0 ARMA_HDF5_INCLUDE_DIR)
++    if(NOT USE_SYSTEM_HDF5)
++      list(GET HDF5_INCLUDE_DIRS 1 ARMA_HDF5_INCLUDE_DIR)
++    else()
++      list(GET HDF5_INCLUDE_DIRS 0 ARMA_HDF5_INCLUDE_DIR)
++    endif()
++
+     message(STATUS "ARMA_HDF5_INCLUDE_DIR = ${ARMA_HDF5_INCLUDE_DIR}")
+     message(STATUS "")
+     message(STATUS "*** If use of HDF5 is causing problems,")
+@@ -480,7 +529,11 @@ if(DETECT_HDF5)
+   endif()
+ endif()
+ 
+-include(ARMA_FindARPACK)
++if(USE_SYSTEM_ARPACK)
++  include(ARMA_FindARPACK)
```

Comment:
I am not sure how the provided find package works but this looks offly suspicious are first glance as an option.
Typically consumers using Conan like to avoid "system" packages because they can be fragile.

Can you explain the motivation for adding these options?

Reply:
The find package is provided by armadillo, and the unmodified recipe uses it by default. It looks in a number of common library paths and `${CMAKE_SYSTEM_LIBRARY_PATH}`. The primary reason it's here is to provide the user with autonomy to decide whether they're happy to accept the fragility you describe - all system library options are disabled by default.

It's worth noting that since we're missing essentially all of the optional dependencies for armadillo in CCI, armadillo isn't particularly useful unless we allow the use of system libraries, as the ability to plug in different math kernels to tweak performance to user requirements is where armadillo gets a lot of it's value from - it provides an abstraction over these math kernels.

I would say this should be removed when we have the appropriate packages available in CCI - there are TODOs listed in conanfile.py:147-151 with links to the appropriate issues.

In my view this is relatively low risk since a user would have to opt in to use a system library. If we remove this entirely, we would just be robbing them the autonomy to decide what they're happy to accept in terms of reproducibility.

Reply:
I agree... It's a short term pain for us to maintain this but since we can not provide those deps the users has this optional method.

Could we maybe have a warning/deprecation for those values?

"This option blah_blah_system uses armadillos default dependency search and will be replaced when the package becomes available in ConanCenter"

Reply:
Yep I like the deprecation notice idea. Is this best placed in `validate()`?

Reply:
Thats a safe bet